### PR TITLE
added atp flag

### PR DESCRIPTION
**Pull request by johnnieskywalker**
# [UI | Auto-Take Profit Banner](https://app.shortcut.com/oazo-apps/story/5512/ui-auto-take-profit-banner)

  
## Changes 👷‍♀️
- Added ATP Banner 
image

- Added dropdown
image

  
## How to test 🧪
  
- Check if banner displays correctly as in figma design
- Check if auto take profit is displayed in dropdown, if states of form change correctly

### CODE REVIEW

File: components/Banner.tsx [hunk 1]
```diff
@@ -10,6 +10,7 @@ export const bannerGradientPresets: BannerGradientPresetsArray = {
   autoSell: ['#fef1e1', '#fef5d6'],
   constantMultiply: ['#ffdde7', '#ffe7f5'],
   stopLoss: ['#ffeaea', '#fff5ea'],
+  autoTakeProfit: ['#e0e7ff', '#fae2fc'],
```
Comment (piotrkonowrocki) [inline components/Banner.tsx:13]: This duplicates `autoBuy` gradient - maybe rename `autoBuy` or use it? I wonder which one is less confusing, using incoherent name or duplicating an array.

File: features/automation/optimization/common/controls/OptimizationDetailsControl.tsx [hunk 2]
```diff
@@ -8,6 +8,8 @@ import { VaultHistoryEvent } from 'features/vaultHistory/vaultHistory'
 import { useFeatureToggle } from 'helpers/useFeatureToggle'
 import React from 'react'
 
+import { AutoTakeProfitDetailsControl } from '../../autoTakeProfit/controls/AutoTakeProfitDetailsControl'
```
Comment (piotrkonowrocki) [inline features/automation/optimization/common/controls/OptimizationDetailsControl.tsx:11]: Relative imports 🤨

File: public/locales/en/common.json [hunk 3]
```diff
@@ -1555,6 +1556,17 @@
...
+  "auto-take-profit": {
+    "title": "Auto-Take Profit",
+    "cancel-summary-title": "Cancel Auto-Take Profit order summary",
+    "form-title": "Auto-Take Profit Setup",
+    "banner": {
+      "header": "Set up Auto-Take Profit",
+      "content": "Auto-Take Profit allows you to set up an exit strategy that will automatically close your vault and take profits when your collateral hits your target price.",
+      "button": "Setup Auto-Take Profit"
+    }
+    },
```
Comment (piotrkonowrocki) [inline public/locales/en/common.json:1568]: Something off with formatting here.

**Review by piotrkonowrocki — CHANGES_REQUESTED** [review]

**Comment by piotrkonowrocki** [comment]
Looks fine, tho.

image

File: components/Banner.tsx (hunk 1 above)
Comment (johnnieskywalker) [reply components/Banner.tsx:13]: On one hand yes, on other it's easier if sb wanted to change gradient of just autoTake or autoBuy

File: features/automation/optimization/common/controls/OptimizationDetailsControl.tsx (hunk 2 above)
Comment (johnnieskywalker) [reply features/automation/optimization/common/controls/OptimizationDetailsControl.tsx:11]: sorry IDE's fault

File: public/locales/en/common.json [hunk 4]
```diff
@@ -1555,6 +1556,17 @@
     "sl-too-high": "Auto-Buy is not available as your existing Stop-Loss Protection trigger is too high. Please <0>adjust your Stop-Loss Trigger</0> below {{maxStopLoss}}% before using Auto-Buy.",
     "coll-ratio-too-high": "Auto-Buy is not available as your Collateral Ratio is too high. Please <0>lower your collateral ratio</0> below {{maxAutoBuyCollRatio}}% before using Auto-Buy."
   },
+  "auto-take-profit": {
+    "title": "Auto-Take Profit",
+    "cancel-summary-title": "Cancel Auto-Take Profit order summary",
+    "form-title": "Auto-Take Profit Setup",
+    "banner": {
+      "header": "Set up Auto-Take Profit",
+      "content": "Auto-Take Profit allows you to set up an exit strategy that will automatically close your vault and take profits when your collateral hits your target price.",
```
Comment (piekczyk) [inline public/locales/en/common.json:1565]: remove `.` from end

**Review by piekczyk — CHANGES_REQUESTED** [review]

File: public/locales/en/common.json (hunk 3 above)
Comment (johnnieskywalker) [reply public/locales/en/common.json:1568]: ah strange that formatter didnt fixed it

**Review by piotrkonowrocki — APPROVED** [review]

**Review by piekczyk — APPROVED** [review]

**Pull request merged**